aws/internal/keyvaluetags/generators/updatetags/main.go

```diff
@@ -188,6 +191,33 @@ func {{ . | Title }}UpdateTags(conn {{ . | ClientType }}, identifier string{{ if
 	newTags := New(newTagsMap)
 
 	if removedTags := oldTags.Removed(newTags); len(removedTags) > 0 {
+		{{- if . | TagFunctionBatchSize }}
+		chunks := removedTags.Chunks({{ . | TagFunctionBatchSize }})
+
+		for _, chunk := range chunks {
+			input := &{{ . | TagPackage }}.{{ . | UntagFunction }}Input{
+				{{- if . | TagInputIdentifierRequiresSlice }}
+				{{ . | TagInputIdentifierField }}:   aws.StringSlice([]string{identifier}),
+				{{- else }}
+				{{ . | TagInputIdentifierField }}:   aws.String(identifier),
+				{{- end }}
+				{{- if . | TagInputResourceTypeField }}
+				{{ . | TagInputResourceTypeField }}: aws.String(resourceType),
+				{{- end }}
+				{{- if . | UntagInputRequiresTagType }}
+				{{ . | UntagInputTagsField }}:       chunk.IgnoreAws().{{ . | Title }}Tags(),
+				{{- else }}
+				{{ . | UntagInputTagsField }}:       aws.StringSlice(chunk.Keys()),
+				{{- end }}
+			}
+
+			_, err := conn.{{ . | UntagFunction }}(input)
+
+			if err != nil {
+				return fmt.Errorf("error untagging resource (%s): %w", identifier, err)
+			}
+		}
+		{{- else }}
```

Comment:
Potential optimization: Given this nice `KeyValueTags.Chunks()` implementation (💯), we might be able to skip redeclaring the majority of the copy-paste template content by re-scoping the `removedTags` and `updatedTags` variables in the loop:

```go
		{{- if . | TagFunctionBatchSize }}
		for _, removedTags := range removedTags.Chunks({{ . | TagFunctionBatchSize }}) {
		{{- end }}
```

The template generator should automatically fix up the formatting whether it should be nested for the looping or not. This doesn't seem worth holding up this merge though. 😄 

We could also support always calling `for _, removedTags := range removedTags.Chunks(-1)` for other services and forego the template conditional, but only adding the looping logic when batching is enabled seems more right.

Reply:
PR submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/11502

Reply:
@bflad Nice,
I didn't know about the formatting fix up with the template generator and had thought about something like:
```go
{{- if . | TagFunctionBatchSize }}
chunks := removedTags.Chunks({{ . | TagFunctionBatchSize }})
{{- else }}
chunks := []KeyValueTags{removedTags}
```